### PR TITLE
[AOTI] Relax input alignment assertion

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #143236

Summary: https://github.com/pytorch/pytorch/pull/142136 added a runtime alignment assertion. But the assumption is probably too strict for more flexible use cases of AOTI, e.g. python deployment, see a recent error torchchat ran into for more details, https://github.com/pytorch/torchchat/actions/runs/12322072267/job/34394851280 . This PR relaxes the runtime check and implements copy_misaligned_inputs in cpp instead.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @chauhang @aakhundov

Differential Revision: [D67287922](https://our.internmc.facebook.com/intern/diff/D67287922)

### CODE REVIEW

File: torch/_inductor/codegen/cpp_wrapper_gpu.py [hunk 1]
```diff
@@ -247,7 +244,9 @@ def codegen_inputs(self):
                 self.prefix.splice(
                     f"""
                     if ((long({input_name}.data_ptr()) & ({GPU_ALIGN_BYTES} -1)) != 0) {{
-                        throw std::runtime_error("{input_name} is not aligned to {GPU_ALIGN_BYTES} bytes");
+                        AtenTensorHandle {input_name}_aligned;
```
Comment (malfet) [inline torch/_inductor/codegen/cpp_wrapper_gpu.py:247]: Should we warn here? Because misaligned inputs can significantly affect model performance

**Review by malfet — APPROVED** [review]

File: torch/_inductor/codegen/cpp_wrapper_gpu.py (hunk 1 above)
Comment (desertfire) [reply torch/_inductor/codegen/cpp_wrapper_gpu.py:253]: Warning added

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D67287922).

**Review by chenyang78 — APPROVED** [review]

**Comment by mikekgfb** [comment]
I think most tensors are allocated aligned, but slicing and views may break that in Python/Pytorch. One solution might be to literally clone every input in flat_inputs. And run a few benchmarks?  How are views and slices dealt with in Inductor?  Unless there's a copy to make these aligned, the same issue would happen with slices generated with inductor, or when calling from byte code into JIT-compiled Inductor code? Did adding assertions actually improve code quality/speed? 

The right place may be just around the actual control transfer to C/C++ -- you'd want to do this in the Python caller (or a C/C++ service function that clones all Pytorch arrays as arguments, optionally only *if and only if* they need a copy ). Also, If we push it in the callee, then every caller, even those that already provide aligned arguments will pay. (Or at least have to evaluate an alignment check and conditional...)

Either way, the payoff may be significant for large matrices -- copy is O(n^2), matmul O(n^3) for a [n,n] *[n,n] GEMM.
(Some BLAS libraries do a copy of input arguments proactively to have nice alignment etc.)

**Pull request closed without merging**